src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/env.py

```diff
@@ -240,8 +225,33 @@ def env_cluster(force_connection, forwarded_port, verb, context=CommandLineInter
             print('[Debug] Resetting.')
         conf = {}
 
+    if not context.env_is_k8s:
+        # if -f was specified, try direct connection only
+        if force_connection:
+            (acs_is_setup, port) = validate_acs_marathon(context, 0)
+        # if only -p specified, without a port number, set up a new tunnel.
+        elif not forwarded_port:
+            (acs_is_setup, port) = acs_marathon_setup(context)
+        # if either no arguments specified (forwarded_port == -1), or -p NNNNN specified (forwarded_port == NNNNN),
+        # test for an existing connection (-1), or the specified port (NNNNN)
+        elif forwarded_port:
+            (acs_is_setup, port) = validate_acs_marathon(context, forwarded_port)
+        # This should never happen
+        else:
+            (acs_is_setup, port) = (False, -1)
+
+        if not acs_is_setup:
+            continue_without_acs = context.get_input(
+                'Could not connect to ACS cluster. Continue with cluster mode anyway (y/N)? ')
+            continue_without_acs = continue_without_acs.strip().lower()
+            if continue_without_acs != 'y' and continue_without_acs != 'yes':
+                print(
+                    "Aborting switch to cluster mode. Please run 'az ml env about' for more information on setting up your cluster.")  # pylint: disable=line-too-long
+                return
+
+        conf['port'] = port
+
```

Comment:
Add call to az_get_k8s_credentials here.

We will need to get the user's resource group but could just form it from one of the existing environment variables minus a suffix.